pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -3673,9 +3675,39 @@ var _ = Context("Scheduling", func() {
 					g.Expect(lo.FromPtr(m.Gauge.Value)).To(BeNumerically(">", 0))
 				}, time.Second).Should(Succeed())
 			}()
-			s.Solve(injection.WithControllerName(ctx, "provisioner"), pods)
+			s.Solve(injection.WithControllerName(ctx, "provisioner"), pods, scheduling.MaxInstanceTypes)
 			wg.Wait()
 		})
+		It("should surface the UnschedulablePodsCount metric while executing the scheduling loop", func() {
+			nodePool := test.NodePool()
+			nodePool.Spec.Template.Spec.Requirements = []v1.NodeSelectorRequirementWithMinValues{
+				{NodeSelectorRequirement: corev1.NodeSelectorRequirement{Key: corev1.LabelInstanceTypeStable, Operator: corev1.NodeSelectorOpIn, Values: []string{"default-instance-type"}}}}
+			ExpectApplied(ctx, env.Client, nodePool)
+			//Creates 15 pods, 5 schedulable and 10 unschedulable and adds and UID
```

Comment:
Could you elaborate on why you need to add the UID here?

Reply:
When the pod fails to schedule it is pushed back onto the queue and the length of the queue at the time it is pushed is mapped to the pod via its UID. If you do not set it in the test the length is mapped to every pod and not just the pod that failed to schedule because they all have an empty UID. That is what caused the issue we were discussing Friday.

Reply:
Can you leave something to that effect as a comment? It's not clear looking at the test why it's necessary.